tools/bouncer_worker/src/lib/queueHandler.js

```diff
@@ -54,22 +92,16 @@ const listenToQueue = (channel, queueName, prefetchCount, callback) => {
 	}, { noAck: false });
 };
 
-const establishChannel = async (conn, listenToJobQueue, listenToModelQueue) => {
+const establishChannel = async (conn, queueNames) => {
 	const channel = await conn.createChannel();
 	channel.assertQueue(rabbitmq.callback_queue, { durable: true });
-	if (listenToJobQueue) {
-		if (!JobQHandler.validateConfiguration(logLabel)) {
-			exitApplication();
-		}
-		listenToQueue(channel, rabbitmq.worker_queue, rabbitmq.task_prefetch, JobQHandler.onMessageReceived);
-	}
-
-	if (listenToModelQueue) {
-		if (!ModelQHandler.validateConfiguration(logLabel)) {
+	queueNames.forEach(queueName => {
+		const handler = queueHandlers[queueName];
+		if(!handler.validateConfiguration(logLabel)) {
 			exitApplication();
 		}
-		listenToQueue(channel, rabbitmq.model_queue, rabbitmq.model_prefetch, ModelQHandler.onMessageReceived);
-	}
+		listenToQueue(channel, queueName, rabbitmq.task_prefetch, handler.onMessageReceived);
```

Comment:
Nice that it's in a loop instead of copy and pasting, but using `task_prefetch` everywhere will be problematic.

typically `task_prefetch` is set to 4 (we process up to 4 federation edits at any one time), but `model_prefetch` is set to 1 (only ever run 1 model import at any one time)

Reply:
Hi @carmenfan , oops 🤦, I have updated it so the prefetch count comes from handler, so each `Handler` can get the correct entry from the config.

Do you want a new config entry for drawings, or just for it to use `model_prefetch` for now?

Reply:
yes if we can add one for drawing so it's all configurable that'll be good, thanks!

Reply:
Done, the new member is called `drawing_prefetch`, e.g.

```
	"rabbitmq": {
		"host": "amqp://bouncerWorker:bouncerWorker@rabbitmq:5672",
		"worker_queue": "jobq",
		"callback_queue": "callbackq",
		"model_queue": "modelq",
		"drawing_queue": "drawingq",
		"task_prefetch": 4,
		"model_prefetch": 1,
		"drawing_prefetch": 2,
		"sharedDir": "D:/3drepo/local/efs/queue"
	},
```